### PR TITLE
Add tests for use of unassigned locals in scripts

### PR DESCRIPTION
**Pull request by cston**

### CODE REVIEW

**Comment by cston** [comment]
@gafter

File: src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs [hunk 1]
```diff
@@ -717,6 +716,56 @@ public void ERR_VariableUsedBeforeDeclaration()
                 Diagnostic(ErrorCode.ERR_UseDefViolation, "x").WithArguments("x").WithLocation(1, 22));
         }
 
+        [WorkItem(550)]
+        [Fact]
+        public void ERR_VariableUsedBeforeDeclaration_02()
+        {
+            var c = CreateSubmission(
+@"object b = a;
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs]: If we ever want to converge "standard" C# with the interactive variant, we'd have to get LDM signoff on this not being an error.

File: src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs [hunk 2]
```diff
@@ -717,6 +716,56 @@ public void ERR_VariableUsedBeforeDeclaration()
...
+                Diagnostic(ErrorCode.ERR_VariableUsedBeforeDeclaration, "c").WithArguments("c").WithLocation(5, 16));
+        }
+
+        [WorkItem(550)]
+        [Fact]
+        public void ERR_UseDefViolation()
+        {
+            var c = CreateSubmission(
+@"int a;
+int b = a;
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/ScriptSemanticsTests.cs]: This too.

**Comment by gafter** [comment]
LGTM

**Pull request merged**